pyccel/codegen/printing/fcode.py

```diff
@@ -1860,24 +1862,85 @@ def _do_range(target, iterable, prolog, epilog):
     #                   OpenMP statements
     # .....................................................
     def _print_OMP_Parallel_Construct(self, expr):
+        clauses = ''
+        if expr.combined:
+            combined = expr.combined.replace("for", omploop_dict["for"])
+            clauses = ' ' + combined
+        clauses += str(expr.txt)
+        omp_expr = '!$omp parallel{}\n'.format(clauses)
+        return omp_expr
+
+    def _print_OMP_For_Loop(self, expr):
         omp_expr   = str(expr.txt)
-        ompexpr = '!$omp {}\n'.format(omp_expr)
-        return ompexpr
+        return '!$omp do{}\n'.format(omp_expr)
 
-    def _print_Omp_End_Clause(self, expr):
+    def _print_OMP_Single_Construct(self, expr):
+        omp_expr   = str(expr.txt)
+        omp_expr = '!$omp {}\n'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Critical_Construct(self, expr):
+        omp_expr = str(expr.txt)
+        omp_expr = '!$omp {}\n'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Master_Construct(self, expr):
+        omp_expr = str(expr.txt)
+        omp_expr = '!$omp {}\n'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Masked_Construct(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr = '!$omp masked{}\n'.format(clauses)
+        return omp_expr
+
+    def _print_OMP_Cancel_Construct(self, expr):
         omp_expr = str(expr.txt)
         omp_expr = omp_expr.replace("for", "do")
-        ompexpr = '!$omp {}\n'.format(omp_expr)
-        return ompexpr
+        omp_expr = '!$omp {}\n'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Target_Construct(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined.replace("for", "do")
+        clauses += str(expr.txt)
```

Comment:
this can be simplified by: clauses = expr.combined.replace("for", "do") + str(expr.txt) if expr.combined else str(expr.txt)